public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -517,15 +534,33 @@ export class LokiDatasource
   }
 
   getLogRowContext = (row: LogRowModel, options?: RowContextOptions): Promise<{ data: DataFrame[] }> => {
-    const target = this.prepareLogRowContextQueryTarget(
-      row,
-      (options && options.limit) || 10,
-      (options && options.direction) || 'BACKWARD'
-    );
+    const direction = (options && options.direction) || 'BACKWARD';
+    const { query, range } = this.prepareLogRowContextQueryTarget(row, (options && options.limit) || 10, direction);
+
+    const reverseResultIfNeeded = (result: DataQueryResponse): DataQueryResponse => {
```

Comment:
Just curious - why did you decide to create this reverse function within `getLogRowContext` and not as datasource method. I am thinking that this could be valuable in the future if we decide to implement query direction feature and this could be harder to find.

Reply:
i see your point. for now i would recommend to keep it where it is, and extract into a helper-function at that point. reason being, i'm not sure if this approach would be used with query-direction (i get confused when both loki-query-direction and reverse-thing happens at the same time 😄 )